util/chplenv/chpl_compiler.py

```diff
@@ -3,9 +3,17 @@
 import os
 import sys
 
-from distutils.spawn import find_executable
-
-import chpl_platform, chpl_locale_model, overrides
+try:
+    # Module `distutils` is deprecated in Python 3.10 and will be removed in Python 3.12
+    # Prefer `shutil.which` in Python 3.2+
+    from shutil import which
+except ImportError:
+    # Backport for pre Python 3.2
+    from distutils.spawn import find_executable as which
+
+import chpl_platform
+import chpl_locale_model
+import overrides
```

Comment:
I suspect a new line for each import is preferred by PEP8 (and done by an auto formatting tool). We have tended to use a single line for importing from other files in this directory, so I have a minor preference for sticking with that to be consistent with other scripts in dir.